users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -75,14 +85,28 @@ public UserManagerService(TransportCreateUserAction transportCreateUserAction,
 
     @Override
     public CompletableFuture<Long> createUser(String userName) {
-        FutureActionListener<WriteUserResponse, Long> listener = new FutureActionListener<>(r -> 1L);
+        FutureActionListener<WriteUserResponse, Long> listener = new FutureActionListener<>(r -> {
+            if (r.doesUserExist()) {
+                throw new UserAlreadyExistsException(userName);
+            }
+            return 1L;
+        });
         transportCreateUserAction.execute(new CreateUserRequest(userName), listener);
         return listener;
     }
 
     @Override
     public CompletableFuture<Long> dropUser(String userName, boolean ifExists) {
-        FutureActionListener<WriteUserResponse, Long> listener = new FutureActionListener<>(WriteUserResponse::affectedRows);
+        DROP_USER_NOT_SUPERUSER.accept(findUser(userName), userName);
```

Comment:
The `user` already has a `name()` property, why is it necessary to pass the userName as argument as well?